src/fsw/FCCode/AttitudeController.cpp

```diff
@@ -0,0 +1,236 @@
+        gnc::utl::rotate_frame(q_body_ecef, v); // v = v_body_0
+        gnc::utl::dcm(DCM_hill_body, r, v);     // Calculate our DCM
+
+        lin::Vector3f dr = pos_baseline_ecef_fp->get(); // dr = dr_ecef
+
+        // Ensure we have a valid relative position
+        if (lin::all(lin::isfinite(dr))) {
+            dr_body = dr;                            // dr_body = dr_ecef
+            gnc::utl::rotate_frame(q_body_ecef, dr_body); // dr_body = dr_body
+        }
+    }
+
+    adcs_state_t state = static_cast<adcs_state_t>(adcs_state_fp->get());
+    switch (state) {
+        /*
+         * The general idea for standby pointing is to have the antenna face
+         * along the velocity vector (we're assuming a near circular orbt here)
+         * and have the docking face pointing normal to our orbit.
+         */
+        case adcs_state_t::point_standby:
+            // Ensure we have a DCM and time
+            if (lin::any(!lin::isfinite(DCM_hill_body)))
+                return;
+            pointer_vec1_current_f.set({1.0f, 0.0f, 0.0f}); // Antenna face
+            pointer_vec2_current_f.set({0.0f, 0.0f, 1.0f}); // Docking face
+            pointer_vec1_desired_f.set(lin::transpose(lin::ref_row(DCM_hill_body, 1))); // v_hat_body
+            pointer_vec2_desired_f.set(lin::transpose(lin::ref_row(DCM_hill_body, 2))); // n_hat_body
+            break;
+        /*
+         * Here we simply want to point the docking face towards the other
+         * satellte and then try to keep GPS for RTK purposes.
+         */
+        case adcs_state_t::point_docking: {
+            if (lin::any(!lin::isfinite(DCM_hill_body)) || lin::any(!lin::isfinite(dr_body)))
+                return;
+            pointer_vec1_current_f.set({0.0f, 0.0f, 1.0f}); // Docking face
+            pointer_vec2_current_f.set({1.0f, 0.0f, 0.0f}); // Antenna face
+            pointer_vec1_desired_f.set(dr_body / lin::norm(dr_body));   // dr_hat
+            pointer_vec2_desired_f.set(lin::transpose(lin::ref_row(DCM_hill_body, 2))); // n_hat_body
+            break;
+        }
+        /* In other adcs states, we won't specify a pointing strategy.
+         */
+        default:
+            break;
+    }
+}
+
+void AttitudeController::calculate_pointing_controller() {
+    // Default all inputs to NaNs and set appropriate fields
+    pointer_data = gnc::PointingControllerData();
+    pointer_data.primary_desired   = pointer_vec1_desired_f.get();
+    pointer_data.primary_current   = pointer_vec1_current_f.get();
+    pointer_data.secondary_desired = pointer_vec2_desired_f.get();
+    pointer_data.secondary_current = pointer_vec2_current_f.get();
+    pointer_data.w_wheels          = w_wheels_rd_fp->get();
+    pointer_data.w_sat             = w_body_est_fp->get();
+    pointer_data.b                 = b_body_rd_fp->get();
+
+    if (lin::any(!(lin::isfinite(pointer_data.primary_desired) && lin::isfinite(pointer_data.primary_current))))
```

Comment:
I know I wrote these initially but reading through them again they're a little hard to follow... This may be cleaner:
```suggestion
    if ( !lin::all(lin::isfinite(pointer_data.primary_desired)) || !lin::all(lin::isfinite(pointer_data.primary_current)))
```
totally up to you though.